web/ui/react-app/src/Checkbox.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { FC, HTMLProps, memo } from 'react'
+import { FormGroup, Label, Input } from 'reactstrap'
+import { uuidGen } from './utils/func'
+
+const Checkbox: FC<HTMLProps<HTMLSpanElement>> = ({ children, onChange, style }) => {
+  const id = uuidGen()
+  return (
+    <FormGroup className="custom-control custom-checkbox" style={style}>
+      <Input
+        onChange={onChange}
+        type="checkbox"
+        className="custom-control-input"
+        id={`checkbox_${id}`}
+        placeholder="password placeholder" />
+      <Label style={{ userSelect: 'none' }} className="custom-control-label" for={`checkbox_${id}`}>
+        {children}
+      </Label>
+    </FormGroup>
+  )
+}
+
+export default memo(Checkbox)
```

Comment:
nit: missing newline :)

Hmm, something with the memo-ing is off. Toggling the checkbox causes the random id to change due to some re-render. I guess that's not what you intended, even though it probably doesn't do much harm?

Reply:
Actually I knew about that and I was thinking to make it a class component in order to preserve the id but as you notice, it's not going to be a problem.